### PR TITLE
feat(web): turn on nginx gzip function

### PR DESCRIPTION
**Pull request by ylduang**

### CODE REVIEW

**Review by hyrious — DISMISSED** [review]

File: web/flat-web/deploy/nginx/nginx.conf [hunk 1]
```diff
@@ -15,6 +15,19 @@ http {
     listen 80;
     root /app;
 
+    gzip on;
+    gzip_disable "MSIE [1-6]\.";
+    gzip_min_length 64;
+    gzip_buffers 16 8k;
+    gzip_comp_level 6;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]:
```suggestion
    gzip_comp_level 1;
```

Just keep it at 1, 6 will put a lot of stress on the CPU.

File: web/flat-web/deploy/nginx/nginx.conf [hunk 2]
```diff
@@ -15,6 +15,19 @@ http {
...
+    gzip_static on;
+    gzip_http_version 1.1;
+    gzip_vary on;
+    gzip_proxied expired no-cache no-store private auth;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]:
```suggestion
```

Keep default

File: web/flat-web/deploy/nginx/nginx.conf [hunk 3]
```diff
@@ -15,6 +15,19 @@ http {
...
+    gzip_types application/javascript application/x-javascript application/json application/manifest+json application/xhtml+xml application/xml video/mp4 font/eot font/otf font/ttf image/svg+xml image/jpeg image/gif image/png text/css text/javascript text/plain text/xml;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]:
```suggestion
    gzip_types application/javascript application/x-javascript text/javascript;
```

We only need `js` (Current stage)

File: web/flat-web/deploy/nginx/nginx.conf [hunk 4]
```diff
@@ -15,6 +15,19 @@ http {
...
+    gunzip on;
+    gunzip_buffers 16 8k;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]:
```suggestion
```

Keep default

File: web/flat-web/deploy/nginx/nginx.conf [hunk 5]
```diff
@@ -15,6 +15,19 @@ http {
     listen 80;
     root /app;
 
+    gzip on;
+    gzip_disable "MSIE [1-6]\.";
+    gzip_min_length 64;
+    gzip_buffers 16 8k;
+    gzip_comp_level 6;
+    gzip_static on;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]: We need enable `--with-http_gzip_static_module`. See: http://nginx.org/en/docs/http/ngx_http_gzip_static_module.html#gzip_static

**Review by BlackHole1 — CHANGES_REQUESTED** [review]
Perhaps we only need `gzip_static`? 🤔

File: web/flat-web/deploy/nginx/nginx.conf (hunk 5 above)
Comment (BlackHole1) [reply web/flat-web/deploy/nginx/nginx.conf]: It looks like we don't need to make any additional changes, the `Nginx` image carries it by default.

image

Ref: https://github.com/nginxinc/docker-nginx/issues/14#issuecomment-71465684

Comment (ylduang) [reply web/flat-web/deploy/nginx/nginx.conf]: It seems that the default startup configuration of nginx on the server is already very nice

File: web/flat-web/deploy/nginx/nginx.conf [hunk 6]
```diff
@@ -18,6 +18,9 @@ http {
     location ~ ^/assets.+$ {
         add_header Cache-Control "public, max-age=31536000";
         try_files $uri 404;
+        gzip on;
+        gzip_static on;
+        gzip_types application/javascript application/x-javascript text/javascript;
```
Comment (BlackHole1) [inline web/flat-web/deploy/nginx/nginx.conf]:
```suggestion
        gzip on;
        gzip_static on;
        gzip_types application/javascript application/x-javascript text/javascript;

        add_header Cache-Control "public, max-age=31536000";
        try_files $uri 404;
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]
nit

**Review by BlackHole1 — DISMISSED** [review]
LGTM.
@ooeyuna Request review again

**Review by ooeyuna — DISMISSED** [review]

**Comment by BlackHole1** [comment]
Waiting for branch [classroom-refactor](https://github.com/netless-io/flat/tree/classroom-refactor) to be merged first.

**Comment by BlackHole1** [comment]
@ASmallBean Can you resolve conflicts? 🙇‍♂️

**Review by BlackHole1 — APPROVED** [review]

**Pull request merged**